pandas/tests/scalar/test_na_scalar.py

```diff
@@ -0,0 +1,108 @@
+import numpy as np
+import pytest
+
+from pandas.core.na_scalar import NA
+
+
+def test_singleton():
+    assert NA is NA
+    new_NA = type(NA)()
+    assert new_NA is NA
+
+
+def test_repr():
+    assert repr(NA) == "NA"
+    assert str(NA) == "NA"
+
+
+def test_truthiness():
+    with pytest.raises(TypeError):
+        bool(NA)
+
+    with pytest.raises(TypeError):
+        not NA
+
+
+def test_hashable():
+    assert hash(NA) == hash(NA)
+    d = {NA: "test"}
+    assert d[NA] == "test"
+
+
+def test_arithmetic_ops(all_arithmetic_functions):
+    op = all_arithmetic_functions
+
+    for other in [NA, 1, 1.0, "a", np.int64(1), np.nan]:
+        if op.__name__ == "rmod" and isinstance(other, str):
+            continue
+        assert op(NA, other) is NA
```

Comment:
I'm not 100% sure of this, but do you also need to test the reverse ops, i.e., add
```
                      assert op(other, NA) is NA
```

Reply:
I think they are included in `all_arithmetic_functions`

Reply:
That's why I said I'm not 100% sure.  I know that when I worked on the ops for `ExtensionArray`, I discovered that the way I was leveraging tests for the operators wasn't properly testing the reverse operators, but that was over a year ago, and it involves a different set of tests, so maybe we're OK.